src/App.js

```diff
@@ -1,16 +1,85 @@
-import React from 'react';
+import React, {useState} from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import ColorChoice from './components/ColorChoice';
 
 const App = () => {
+  const[entriesData, setEntriesData] = useState(chatMessages);
+
+  const likeEntry = (id) => {
+    setEntriesData((prev)=>{
+      return prev.map((entry)=>{
+        if(id === entry.id){
+          return{
+            ...entry,
+            liked: !entry.liked
+          };}
+        else{
+          return entry;
+        }
+        });
+    });
+  };
+
+  const totalLikes= entriesData.reduce((total, entry)=>{
+    if(entry.liked === true){
+      total +=1;
+    }
+    return total;
+  }, 0);
+
+  const [senderColors, setSenderColors] = useState({
+    [chatMessages[0].sender]: 'black',
+    [chatMessages[1].sender]: 'black'
+  });
+
+  const handleSenderColor = (sender, color) => {
+    setSenderColors((prevColors) => ({
+      ...prevColors,
+      [sender]: color
+    }));
+    
+    //I updated the data to set the text color dinamically
+    const updatedEntriesData = entriesData.map((entry) => {
+      if (entry.sender === sender) {
+        return {
+          ...entry,
+          color: color
+        };
+      } else {
+        return entry;
+      }
+    });
+  
+    setEntriesData(updatedEntriesData);
```

Comment:
We could use an approach like this to add addiitonal keys into ur data, but it does end up being a little repetitious. Even if we wanted to pass the color explicitly to the ChatEntries, we could look up the color in ChatLog to pass to ChatEntry. But since you went with using the `senderColors` data in ChatEntry directly, this could be removed now.